todo_app/app.py

```diff
@@ -1,92 +1,223 @@
 
 from todo_app.data.todo_mongo_client import *
 from todo_app.data.viewmodel import *
-from flask import Flask, render_template, request, redirect
+from todo_app.data.user import *
+from flask import Flask, render_template, request, redirect, session
+from flask_login import login_required
+from flask_login import LoginManager
+from flask_login import login_user
+from flask_login import current_user
 import os
+import requests
+import json
+from oauthlib.oauth2 import WebApplicationClient
 
-def create_app(db_name = ""):
+
+
+def create_app():
    app = Flask(__name__)
    item_view_model = None
 
+   disable_login = os.getenv('FLASK_SKIP_LOGIN')
+   if disable_login:
+      print("Switching off authentication in the app config for testing purposes")
+      app.config['LOGIN_DISABLED'] = True
    mongo_srv = os.getenv('MONGO_SRV')
-   if db_name == "":
-      db_name = os.getenv('MONGO_DB')
+   db_name = os.getenv('MONGO_DB')
    mongo_user = os.getenv('MONGO_USER')
    mongo_pwd = os.getenv('MONGO_PWD')
    mongo_connection = os.getenv('MONGO_CONNECTION')
 
    mongo_client = ToDoMongoClient(mongo_user, mongo_pwd, mongo_srv, db_name, mongo_connection)
 
+   oauth_client_id = os.getenv('OAUTH_CLIENT_ID')
+   oauth_secret_id = os.getenv('OAUTH_CLIENT_SECRET')
+
+   login_manager = LoginManager() 
+   login_manager.init_app(app)
+   client = WebApplicationClient(oauth_client_id)
+
+   os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
+   app.secret_key = oauth_secret_id
```

Comment:
The app's secret key is actually not related to the OAuth secret - the former is for signing the session cookie, while the latter is about verifying your app's identity during OAuth so best practice would be to have two separate secrets for these

Reply:
Sorry Jack, not sure I understand this - where should I get the app secret key from then? And what should I do with the oauth secret id?

Reply:
@JackMead what should I put as the app.secret_key?